### DISCUSSION TITLE
Query slow when using bindings

### DISCUSSION THREAD

**Discussion by koneubauer** [Budibase/budibase#12410] [Help]
I created an SQL-Query for my SQL Server instance:

```
SELECT tt.elsmatch + ' - ' + tt.name AS name, tt.image_url, COALESCE(se.count, 0) AS count
FROM em.transporttype tt
LEFT JOIN (
    SELECT tycod, COUNT(*) AS count
    FROM schedevents se
    INNER JOIN em.timeframe tf ON CAST(em.CONVERT_DTS(se.terminzeit) AS TIME) BETWEEN CAST(dbo.CONVERT2DATETIME2CEST(tf.[from]) AS TIME) AND CAST(dbo.CONVERT2DATETIME2CEST(tf.[to]) AS TIME)
    INNER JOIN em.lev3 le ON le.[name] = se.lev3
    WHERE se.dgroup = {{dg}}
        AND CAST(em.CONVERT_DTS(se.terminzeit) AS DATE) = CONVERT(DATE, {{date}})
        AND le.areaid = {{area}}
        AND tf.id = {{tf}}
    GROUP BY se.tycod
) se ON tt.elsmatch = se.tycod;
```
`{{date}}` is a Date in '2023-11-21 format, `{{area}}` and `{{tf}}` are IDs and `{{dg}}` is a string that would for example be '519'.
When I execute the query in budibase and manually write `se.dgrou = '519'` execution takes about 1.5 seconds. When i use `se.dgroup = {{dg}}` execution takes 10 seconds!!

**Comment by ConorWebb96** [comment]
Hey @koneubauer,

- Are you cloud or self-host?
- What version of Budibase are you on?

Does using a different binding name make any difference? (I can't see why it would as i don't theres a protect bindable name that uses `dg`.)

I think we will need a few things from you in this case to debug further as I haven't been able to replicate this locally or on cloud.
- Your database schema for the relevant tables used in the above query
- Test data
- An app export

Hopefully this makes sense.
